### PR TITLE
Make deploy "website" to surge.sh conditional

### PR DESCRIPTION
**Pull request by nktssh**
### 🔦 Summary
Building "website" is now conditional, so Travis CI check would be faster! 🚀🚀🚀 
Refs #800 

### 🗺️ Test Plan
In PR checks, verify that "website" build deployed in this PR (since we changed spelling mistake in website)

### CODE REVIEW

**Comment by nktssh** [comment]
I’ve tried to use my personal token but Github automatically revokes it 🙈

We need to create repository variable with name `GITHUB_TOKEN` and value which will be Github API token with access to scope `repo/public_repo` and `write:discussion`.

File: deploy_website_to_surge.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+if [type = pull_request && ${WEBSITE_CHANGED}]
+then
+  npx surge --project ./build --domain wix-yoshi-${TRAVIS_PULL_REQUEST}.surge.sh;
+
+  #inspired by https://damien.pobel.fr/post/github-api-from-travisci/
+  CURL_RESULT = curl -H "Authorization: token ${GITHUB_TOKEN}" -X POST \
```
Comment (ranyitz) [inline deploy_website_to_surge.sh]: About the message, maybe we can go with something similar to https://github.com/reactjs/reactjs.org/pull/1892#issuecomment-479347823

File: deploy_website_to_surge.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    -d "{\"body\": \"You could check new website at http://wix-yoshi-${TRAVIS_PULL_REQUEST}.surge.sh/yoshi 😉\"}" \
```
Comment (ranyitz) [inline deploy_website_to_surge.sh]:
```suggestion
    -d "{\"body\": \"You could check new website at https://wix-yoshi-${TRAVIS_PULL_REQUEST}.surge.sh/yoshi\"}" \
```

Also I think that you can use html tags to style it a bit 😉

File: .travis.yml [hunk 3]
```diff
@@ -16,6 +16,8 @@ branches:
     - /^version_\d+\.x$/
 cache:
   yarn: true
+env:
+  - WEBSITE_CHANGED = git diff --name-only $TRAVIS_COMMIT_RANGE | grep "website/"
```
Comment (ranyitz) [inline .travis.yml]: Also changes in `/docs` directory are relevant

File: .travis.yml [hunk 4]
```diff
@@ -16,6 +16,8 @@ branches:
...
 jobs:
   include:
     - name: "website"
```
Comment (ranyitz) [inline .travis.yml]:
```suggestion
    - name: "website:publish"
```

File: .travis.yml [hunk 5]
```diff
@@ -25,8 +27,8 @@ jobs:
       - git config --global user.email "${GH_EMAIL}"
       - echo "machine github.com login ${GH_NAME} password ${GH_TOKEN}" > ~/.netrc
       - cd website && yarn install && GIT_USER="${GH_NAME}" yarn run publish-gh-pages 
-    - name: "website - release to surge"
-      if: type = pull_request
+    - name: "website - build for release to surge if there are changes"
```
Comment (ranyitz) [inline .travis.yml]:
```suggestion
    - name: "website:preview"
```

Comment (ranyitz) [reply .travis.yml]: I think that's it's not the best place for a description. We could settle on short names

File: .travis.yml [hunk 6]
```diff
@@ -53,5 +55,5 @@ jobs:
       if: type != pull_request
       before_script: npm install
       script: npm run lint 
-after_success: 
-  - npx surge --project ./build --domain wix-yoshi-${TRAVIS_PULL_REQUEST}.surge.sh
\ No newline at end of file
+after_success:
+  - ./deploy_website_to_surge.sh
```
Comment (ranyitz) [inline .travis.yml]: When thinking about it, I'm not sure that we should preview the site just if the build passes, it means that if you've changed some documentation and want to see how it looks, you need to wait approximately 20 minutes with no reason.

@nktssh what do you think?

File: deploy_website_to_surge.sh (hunk 2 above)
Comment (nktssh) [reply deploy_website_to_surge.sh]: Well, it's so opinionated...
Could you please provide some info how you would like to see the message?

File: .travis.yml (hunk 6 above)
Comment (nktssh) [reply .travis.yml]: Totally agree! It was here so didnt change this part. Moved to jobs

File: .travis.yml (hunk 3 above)
Comment (nktssh) [reply .travis.yml]: Added

File: deploy_website_to_surge.sh (hunk 1 above)
Comment (nktssh) [reply deploy_website_to_surge.sh]: I've changed the comment text a little bit.
Also, we need to register a new Github user and get access token from this account with scopes `repo/public_repo` and `write:discussion`. Those token should be stored in this repository settings under name `GITHUB_TOKEN` (but we could rename it)

File: deploy_website_to_surge.sh (hunk 2 above)
Comment (ranyitz) [reply deploy_website_to_surge.sh]: That's correct, let's start with the message provided and improve later on down the road if we'll feel that's important enough.

File: .travis.yml (hunk 6 above)
Comment (ranyitz) [reply .travis.yml]: Thanks a lot

**Pull request merged**